DeBroglie/Constraints/MaxConsecutiveTilesConstraint.cs

```diff
@@ -0,0 +1,107 @@
+using System;
+using System.Collections.Generic;
+
+namespace DeBroglie.Constraints
+{
+    /// <summary>
+    /// Restrict how many consecutive tiles can contain the target tile on one 
+    /// row of an axis. For instance, you might use this constraint to limit how 
+    /// long a ladder can be in a platformer.
+    /// </summary>
+    public class MaxConsecutiveTilesConstraint : ITileConstraint
+    {
+        private int[] lengths;
+        private int beforeDirection;
+        private int afterDirection;
+
+        public Tile Tile { get; set; }
+
+        public int MaxLength { get; set; }
+
+        public enum AxisType
+        {
+            X,
+            Y,
+            Z,
+        }
+
+        public AxisType Axis { get; set; }
+
+        public Resolution Init(TilePropagator propagator)
+        {
+            var topology = propagator.Topology;
+            
+            switch (Axis)
```

Comment:
I've been trying to avoid enumerating axes, and hard coding the direction indices. It makes it hard to support hexagonal arrays, (and it doesn't fit with some of my other plans). It's better to accept a triple of ints X, Y, Z, then call `Directions.GetDirection` when you need a direction index.

Reply:
I'll move this into my Unity side of things and stick to just directions in your lib.